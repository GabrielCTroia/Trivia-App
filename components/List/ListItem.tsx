import React, { FunctionComponent, ReactElement } from "react";
import { View, StyleSheet, ViewProps, Text } from "react-native";
import * as Colors from "../../styles/Colors";
import * as Effects from "../../styles/Effects";


export type ListItemProps = ViewProps & {
  content: string;
  leftSide?: ReactElement;
  rightSide?: ReactElement;
};

export const ListItem: FunctionComponent<ListItemProps> = ({ leftSide = null, rightSide = null, ...props }) => {
  return (
    <View style={[styles.container, props.style]}>
      {leftSide && <View style={styles.leftSide}>{leftSide}</View>}
      <Text style={styles.content}>
        {props.content}
      </Text>
      {rightSide && <View style={styles.rightSide}>{rightSide}</View>}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    paddingTop: 20,
    paddingBottom: 10,
    borderColor: Colors.lightShade,
    borderBottomWidth: 1,

    flex: 1,
    flexDirection: 'row',
  },

  leftSide: {
    flex: .2,
  },

  content: {
    flex: 1,
    justifyContent: 'center',
    color: Colors.baseTextColor,
  },

  rightSide: {
    flex: .1,
    alignItems: 'flex-end',
    justifyContent: 'center',
  }
});
