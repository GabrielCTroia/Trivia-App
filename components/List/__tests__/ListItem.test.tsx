import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { ListItem } from '../ListItem';
import { Text } from 'react-native';

it('renders correctly', () => {
  const tree = renderer.create(<ListItem content="test" />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with left side', () => {
  const tree = renderer.create(
    <ListItem
      content="test"
      leftSide={<Text>left side test</Text>}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with right side', () => {
  const tree = renderer.create(
    <ListItem
      content="test"
      rightSide={<Text>right side test</Text>}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with all sides', () => {
  const tree = renderer.create(
    <ListItem
      content="test"
      leftSide={<Text>left side test</Text>}
      rightSide={<Text>right side test</Text>}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
