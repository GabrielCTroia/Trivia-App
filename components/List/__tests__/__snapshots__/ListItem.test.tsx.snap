// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E0E0E0",
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#828282",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`renders correctly with all sides 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E0E0E0",
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 0.2,
      }
    }
  >
    <Text>
      left side test
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#828282",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    test
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 0.1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      right side test
    </Text>
  </View>
</View>
`;

exports[`renders correctly with left side 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E0E0E0",
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 0.2,
      }
    }
  >
    <Text>
      left side test
    </Text>
  </View>
  <Text
    style={
      Object {
        "color": "#828282",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`renders correctly with right side 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E0E0E0",
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": 10,
        "paddingTop": 20,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#828282",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    test
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 0.1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      right side test
    </Text>
  </View>
</View>
`;
